aten/src/ATen/native/native_functions.yaml

```diff
@@ -6042,23 +6042,23 @@
     CPU: nll_loss2d_backward_cpu
     CUDA: legacy::cuda::_thnn_nll_loss2d_backward
 
-- func: smooth_l1_loss.out(Tensor self, Tensor target, int reduction=Mean, *, Tensor(a!) out) -> Tensor(a!)
+- func: smooth_l1_loss.out(Tensor self, Tensor target, int reduction=Mean, float beta=1.0, *, Tensor(a!) out) -> Tensor(a!)
```

Comment:
I'm a little confused by the use of floats vs. doubles in native_functions, and how that trickles through everywhere else.

I originally set `beta` as a double, but that caused the codegen portion of the build process to complain (codegen looks like it only accepts a subset of primitives).

I tried to set `beta` as a float in all other function signatures in the other files, but I'm still running into issues. Currently, beta is set as a double in `autocast_mode.cpp` and `FunctionsManual.h`. I get linker errors in each of the test files of the form `undefined reference to 'at::native::smooth_l1_loss(at::Tensor const&, at::Tensor const&, long, double)'`.

When I try changing `beta` to a float in those files to match the other definition, I get compiler errors of the form 
 `error: could not convert template argument ‘at::smooth_l1_loss’ from ‘at::Tensor (*)(..., **double**) {aka at::Tensor (*)(..., **double**)}’ to ‘at::Tensor (*)(..., **float**)’`

I don't think I'm following the codegen workflow for how these files integrate with each other.